### PR TITLE
Docs maintenance

### PR DESCRIPTION
**Pull request by hsz**

### CODE REVIEW

File: basics/plugin_structure/dynamic_plugins.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ title: Dynamic Plugins
 ---
 <!-- Copyright 2000-2020 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file. -->
 
-Starting with 2020.1 release, the ability to install, update and uninstall plugins without restarting the IDE is available in the IntelliJ Platform.
+Starting with the 2020.1 release, installing, updating, and uninstall plugins without restarting the IDE is available in the IntelliJ Platform.
```
Comment (YannCebron) [inline basics/plugin_structure/dynamic_plugins.md:6]: uninstalling

File: basics/plugin_structure/dynamic_plugins.md [hunk 2]
```diff
@@ -18,41 +18,41 @@ will run these checks automatically. See [Plugin Verifier](/reference_guide/api_
...
+The plugin's extension points must adhere to specific usage rules and then [be declared](plugin_extension_points.md#dynamic-extension-points) ready for dynamic use explicitly.
 
 ### Configurables Depending on Extension Points
 
 Any `Configurable` which depends on dynamic extension points must implement `Configurable.WithEpDependencies`.
 
 ### No Use of Service Overrides
 
-Application, project and module services declared with `overrides="true"` are not allowed.
+Application, project, and module services declared with `overrides=" true" ` are not allowed.
```
Comment (YannCebron) [inline basics/plugin_structure/dynamic_plugins.md:43]: why space?

File: basics/plugin_structure/dynamic_plugins.md [hunk 3]
```diff
@@ -18,41 +18,41 @@ will run these checks automatically. See [Plugin Verifier](/reference_guide/api_
...
 
 ## Code
 
-> **NOTE** Loading and unloading plugins happens in EDT and under write action.
+> **NOTE** Loading and unloading plugins happens in EDT and underwrite action.
```
Comment (YannCebron) [inline basics/plugin_structure/dynamic_plugins.md:47]: write action is a term

File: basics/plugin_structure/plugin_content.md [hunk 4]
```diff
@@ -27,7 +27,7 @@ A plugin consisting of a single `.jar` file is placed in the `/plugins` director
 
 
 ### Plugin With Dependencies 
-The plugin `.jar` file is placed in the `/lib` folder under the plugin's "root" folder, together with all required bundled libraries.
+The plugin `.jar` file is placed in the `/lib` folder under the plugin's "root" folder and all required bundled libraries.
```
Comment (YannCebron) [inline basics/plugin_structure/plugin_content.md:30]: IMHO confusing now

File: products/rider.md [hunk 5]
```diff
@@ -6,7 +6,7 @@ title: Rider Plugin Development
 ## Introduction 
 Rider plugins are generally used to expose the functionality of a ReSharper plugin.
 [Rider](https://www.jetbrains.com/rider/) uses the IntelliJ Platform somewhat [differently](/intro/intellij_platform.md#rider) than other Platform-based based IDEs. 
-Rider uses the IntelliJ Platform to provide the user interface for a C# and .NET IDE but uses ReSharper to provide the language-specific features.
+The Rider uses the IntelliJ Platform to provide the user interface for a C# and .NET IDE but uses ReSharper to provide the language-specific features.
```
Comment (YannCebron) [inline products/rider.md:9]: Rider is a product name

File: products/rider.md [hunk 6]
```diff
@@ -19,10 +19,10 @@ Before starting a new Rider plugin project, review the article [Writing plugins
 In particular, this article discusses _One Template to Rule Them All_, a way to quickly get started writing plugins for both Rider and ReSharper. 
 
 More background information is available in the [Building a .NET IDE with JetBrains Rider](https://www.codemag.com/Article/1811091/Building-a-.NET-IDE-with-JetBrains-Rider) article, which includes a high-level description of the _Rider Protocol_ communication process between Rider and ReSharper. 
-The article is a good counterpoint to the ReSharper DevGuide content, which discusses the protocol at the code level.
+The article is an excellent counterpoint to the ReSharper DevGuide content, which discusses the code level protocol.
```
Comment (YannCebron) [inline products/rider.md:22]: meaning of sentence changed

File: products/webstorm.md [hunk 7]
```diff
@@ -25,7 +25,7 @@ To see how these attributes appear in a similar `build.gradle` file for PhpStorm
 
 The dependency on the WebStorm APIs must be declared in the `plugin.xml` file.
 As described in [Modules Specific to Functionality](/basics/getting_started/plugin_compatibility.md#modules-specific-to-functionality) table, the `<depends>` tags must declare `JavaScriptLanguage`.
-**Note** that for WebStorm the plugin.xml file must also declare a dependency on `com.intellij.modules.platform` because `JavaScriptLanguage` is not recognized as a module.
+**Note** that for WebStorm, the plugin.xml file must also declare a dependency on `com.intellij.modules.platformx` because `JavaScriptLanguage` is not recognized as a module.
```
Comment (YannCebron) [inline products/webstorm.md:28]: platformx?

File: reference_guide/api_changes/api_changes_list_2019.md [hunk 8]
```diff
@@ -57,12 +57,12 @@ An example of a ticket is https://youtrack.jetbrains.com/issue/PR-1218. Until su
 
 NOTE: You are allowed to prettify the pattern using links: [`org.example.Foo`](https://github.com/JetBrains/intellij-community/tree/master/)
 
-NOTE: Entries not starting with code quotes (`name`) can be added to document non-code changes and will be skipped in API verification.
+NOTE: Entries not starting with code quotes (`name`) can be added to document non-code changes and be skipped in API verification.
```
Comment (YannCebron) [inline reference_guide/api_changes/api_changes_list_2019.md:58]: no

File: reference_guide/api_changes/api_changes_list_2020.md [hunk 9]
```diff
@@ -57,12 +57,12 @@ An example of a ticket is https://youtrack.jetbrains.com/issue/PR-1218. Until su
 
 NOTE: You are allowed to prettify the pattern using links: [`org.example.Foo`](https://github.com/JetBrains/intellij-community/tree/master/)
 
-NOTE: Entries not starting with code quotes (`name`) can be added to document non-code changes and will be skipped in API verification.
+NOTE: Entries not starting with code quotes (`name`) can be added to document non-code changes and be skipped in API verification.
```
Comment (YannCebron) [inline reference_guide/api_changes/api_changes_list_2020.md:58]: no

File: reference_guide/api_changes/api_changes_list_2020.md [hunk 10]
```diff
@@ -257,7 +257,7 @@ Added Union Types Support
 : Use `com.intellij.testFramework.ReadOnlyLightVirtualFile`-based PSI instead.
 
 Java code migrated to use `TYPE_USE` nullability annotations
-: Due to some problems in Kotlin compiler existing Kotlin code might become incompilable in rare cases if a method written in Java returning an array and annotated as `@Nullable` or `@NotNull` is used or overridden.
+: Due to some problems in Kotlin compiler, existing Kotlin code might become incompatible in rare cases if a method is written in Java returning an array and annotated as `@Nullable` or `@NotNull` is used or overridden.
```
Comment (YannCebron) [inline reference_guide/api_changes/api_changes_list_2020.md:258]: incompilable and incompatible are two distinct things

File: reference_guide/frameworks_and_external_apis/spring_api.md [hunk 11]
```diff
@@ -18,39 +18,39 @@ Please create an IntelliJ Platform SDK to include all minimum required files.
...
-	* `$IDEA_HOME$/lib/src/src_spring-openapi.zip`
+   * `$IDEA_HOME$/lib/src/src_spring-openapi.zip`
 
 ### General Notes
-If you use other Spring functionality (e.g. Spring EL) in your plugin, make sure to add all required JARs to your IntelliJ Platform SDK classpath to make your plugin's tests work.
+If you use other Spring functionality (e.g., Spring EL) in your plugin, add all required JARs to your IntelliJ Platform SDK classpath to make your plugin's tests work.
 
 
-Please use only Spring-related functionality exposed in `spring-api.jar` (where sources are provided) in your plugin. Using any other "internal" (implementation) classes from Spring plugin itself (`spring.jar`) is _not_ supported.
+Please use only Spring-related functionality exposed in `spring-api.jar` (where your plugin is provided). Using any other "internal" (implementation) classes from Spring plugin itself (`spring.jar`) is _not_ supported.
```
Comment (YannCebron) [inline reference_guide/frameworks_and_external_apis/spring_api.md]: wrong

File: tutorials/action_system/grouping_action.md [hunk 12]
```diff
@@ -4,7 +4,7 @@ title: Grouping Actions
 <!-- Copyright 2000-2020 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file. -->
 
 If an implementation requires several actions, or there are simply too many actions that overload the menu, the actions can be placed into groups.
-This tutorial demonstrates adding an action to an existing group, creating a new action group, and action groups with a variable number of actions.
+This tutorial demonstrates adding an action to an existing group, creating a new action group, and action groups with various actions.
```
Comment (YannCebron) [inline tutorials/action_system/grouping_action.md:7]: change of meaning

File: basics/architectural_overview/file_view_providers.md [hunk 13]
```diff
@@ -7,9 +7,9 @@ A file view provider ([`FileViewProvider`](upsource:///platform/core-api/src/com
 
 For example, a JSPX page has a separate PSI tree for the Java code in it (`PsiJavaFile`), a separate tree for the XML code (`XmlFile`), and a separate tree for JSP as a whole ([`JspFile`](upsource:///java/jsp-openapi/src/com/intellij/psi/jsp/JspFile.java)).
 
-Each of the PSI trees covers the entire contents of the file, and contains special "outer language elements" in the places where contents in a different language can be found.
+Each of the PSI trees covers the entire contents of the file and contains unique "outer language elements" in the places where contents in a different language can be found.
```
Comment (YannCebron) [inline basics/architectural_overview/file_view_providers.md:10]: special != unique

File: basics/architectural_overview/general_threading_rules.md [hunk 14]
```diff
@@ -29,16 +29,16 @@ To pass control from a background thread to the event dispatch thread, instead o
...
 : The operation can be executed when the topmost shown dialog is the one that contains the specified component or is one of its parent dialogs. 
 
 None specified
-: `ModalityState.defaultModalityState()` will be used. In most cases, this is the optimal choice, that uses current modality state when invoked from UI thread, and has special handling for background processes started with `ProgressManager`: `invokeLater()` from such a process may run in the same dialog that the process started.
+: `ModalityState.defaultModalityState()` will be used. This is the optimal choice in most cases that uses the current modality state when invoked from UI thread. It has special handling for background processes started with `ProgressManager`: `invokeLater()` from such a process may run in the same dialog that the process started.
 
 `ModalityState.any()`
-: The operation will be executed as soon as possible regardless of modal dialogs. Please note that modifying PSI, VFS, or project model is prohibited from such runnables.
+: The operation will be executed as soon as possible regardless of modal dialogs. Please note that modifying PSI, VFS, or project model is prohibited from such runnable.
```
Comment (YannCebron) [inline basics/architectural_overview/general_threading_rules.md:41]: IMHO plural is correct

File: basics/architectural_overview/general_threading_rules.md [hunk 15]
```diff
@@ -51,19 +51,18 @@ which has plenty of methods to execute the given code with a modal (dialog), non
 In all cases, the code is executed on a background thread, which is associated with a [`ProgressIndicator`](upsource:///platform/core-api/src/com/intellij/openapi/progress/ProgressIndicator.java) object.
 The current thread's indicator can be retrieved any time via `ProgressIndicatorProvider.getGlobalProgressIndicator()`.
 
-For visible progresses, threads can use `ProgressIndicator` to notify the user about current status:
-e.g., set text or visible fraction of the work done.
+For visible progress, threads can use `ProgressIndicator` to notify the user about current status:
```
Comment (YannCebron) [inline basics/architectural_overview/general_threading_rules.md]: plural?

File: basics/getting_started/plugin_compatibility.md [hunk 16]
```diff
@@ -65,13 +65,13 @@ As of this writing, if a plugin: **A)** is dependent _only_ on one or more of th
 ### Modules Specific to Functionality
 More specialized functionality is also delivered via modules and plugins in IntelliJ Platform-based products. 
 For example, the `com.intellij.modules.python` module supports the Python language-specific functionality. 
-If a plugin uses functionality from this module, such as Python-specific inspections and refactoring, it must declare a dependency on this module.
+If a plugin uses this module's functionality, such as Python-specific inspections and refactoring, it must declare a dependency on this module.
 
 Note that not all products define and declare modules. 
-For example, PhpStorm does not have its own module, but the product itself depends on (and ships with) the PHP language plugin. 
+For example, PhpStorm does not have its own module, but its product depends on (and ships with) the PHP language plugin.
```
Comment (YannCebron) [inline basics/getting_started/plugin_compatibility.md]: PhpStorm _is_ the product

File: basics/persisting_state_of_components.md [hunk 17]
```diff
@@ -3,13 +3,13 @@ title: Persisting State of Components
 ---
 <!-- Copyright 2000-2020 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file. -->
 
-The *IntelliJ Platform* provides an API that allows components or services to persist their state between restarts of the IDE. You can use either a simple API to persist a few values or persist the state of more complicated components using the [`PersistentStateComponent`](upsource:///platform/projectModel-api/src/com/intellij/openapi/components/PersistentStateComponent.java) interface.
+The *IntelliJ Platform* provides an API that allows components or services to persist their state between the IDE restarts. You can use either a simple API to persist a few values or persist the state of more complicated components using the [`PersistentStateComponent`](upsource:///platform/projectModel-api/src/com/intellij/openapi/components/PersistentStateComponent.java) interface.
```
Comment (YannCebron) [inline basics/persisting_state_of_components.md]: "between the IDE restarts." grammar?

File: basics/persisting_state_of_components.md [hunk 18]
```diff
@@ -24,13 +24,13 @@ To use it:
 - define the state class
 - specify the storage location using `@com.intellij.openapi.components.State`
 
-Note that instances of extensions cannot persist their state by implementing `PersistentStateComponent`. If your extension needs to have a persistent state, you need to define a separate service responsible for managing that state.
+Note that instances of extensions cannot persist in their state by implementing `PersistentStateComponent`. If your extension needs to have a persistent state, you need to define a separate service responsible for managing it.
```
Comment (YannCebron) [inline basics/persisting_state_of_components.md]: no

File: basics/persisting_state_of_components.md [hunk 19]
```diff
@@ -119,22 +119,22 @@ class State {
...
 ### Defining the Storage Location
 
-To specify where exactly the persisted values are stored, add `@State` annotation to the `PersistentStateComponent` class. 
+To specify where precisely the persisted values are stored, add `@State` annotation to the `PersistentStateComponent` class. 
 
 It has the following fields:
 * `name` (required) — specifies the name of the state (name of the root tag in XML).
-* `storages` — one or more of `@com.intellij.openapi.components.Storage` annotations to specify the storage locations. Optional for project-level values — standard project file is used in this case.
-* `reloadable` (optional) — if set to false, complete project (or application) reload is required when the XML file is changed externally and the state has changed.
+* `storages` — one or more of `@com.intellij.openapi.components.Storage` annotations to determine the storage locations. Optional for project-level values — standard project file is used in this case.
```
Comment (YannCebron) [inline basics/persisting_state_of_components.md]: specify != determine

File: basics/persisting_state_of_components.md [hunk 20]
```diff
@@ -155,17 +155,17 @@ If the underlying persistence model or storage format has changed, a [`Converter
 
 ## Persistent Component Lifecycle
 
-The `loadState()` method is called after the component has been created (only if there is some non-default state persisted for the component), and after the XML file with the persisted state is changed externally (for example, if the project file was updated from the version control system). In the latter case, the component is responsible for updating the UI and other related components according to the changed state.
+The `loadState()` method is called after the component has been created (only if some non-default state persisted for the component). After the XML file with the persisted state is changed externally (for example, if the project file was updated from the version control system). In the latter case, the component is responsible for updating the UI and other related components according to the changed state.
```
Comment (YannCebron) [inline basics/persisting_state_of_components.md]: grammar?

**Review by YannCebron — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 21]
```diff
@@ -103,14 +121,14 @@ git submodule init
...
 ```
 
-Initialization creates a `.gitmodules` file, register a submodule in the `sdkdocs-template` folder and check out the files. 
+Initialization creates a `.gitmodules` file, register a submodule in the `sdkdocs-template` folder and check out the files.
 Note that when a repository is added as a submodule, it doesn't get a `.git` folder, but instead gets a `.git` file that points to the actual location of the `.git` folder.
 
-A submodule can be updated using standard git commands such as `git pull`. 
-It can be switched to a different branch using `git checkout`, and any changes to the currently checked out revision need to be committed back into the main repository using git commands. 
+A submodule can be updated using standard git commands such as `git pull`.
+It can be switched to a different branch using `git checkout`, and any changes to the currently checked out revision need to be committed back into the main repository using git commands.
```
Comment (JohnHake) [inline CONTRIBUTING.md:128]: Superfluous change.

File: CONTRIBUTING.md [hunk 22]
```diff
@@ -103,14 +121,14 @@ git submodule init
...
 A submodule is initially cloned at a specific revision, and not as part of a branch.update
 
-If changes are made to the submodule, they should be made on a branch to a clone, and a Pull Request sent. 
+If changes are made to the submodule, they should be made on a branch to a clone, and a Pull Request sent.
```
Comment (JohnHake) [inline CONTRIBUTING.md:131]: Superfluous change.

File: CONTRIBUTING.md [hunk 23]
```diff
@@ -119,10 +137,9 @@ If there are any problems with the `sdkdocs-template`, please [raise an issue](h
 Content contributions to the IntelliJ Platform SDK are welcome.
 Please download or clone the open-source SDK project from [GitHub](https://github.com/JetBrains/intellij-sdk-docs), make additions or changes, and submit a pull request.
 Before creating or altering content, please consult these guides:
-* [SDK Documentation Style Guide](intro/sdk_style.md). 
+* [SDK Documentation Style Guide](intro/sdk_style.md).
```
Comment (JohnHake) [inline CONTRIBUTING.md:140]: Superfluous change.

File: CONTRIBUTING.md [hunk 24]
```diff
@@ -119,10 +137,9 @@ If there are any problems with the `sdkdocs-template`, please [raise an issue](h
...
   This guide describes documentation conventions in terms of Markdown syntax.
   Always test documentation changes using a [preview](#building-and-previewing-the-site) of the site.
-* [SDK Code Sample Guidelines](intro/sdk_code_guidelines.md). 
+* [SDK Code Sample Guidelines](intro/sdk_code_guidelines.md).
```
Comment (JohnHake) [inline CONTRIBUTING.md:143]: Superfluous change.

File: CONTRIBUTING.md [hunk 25]
```diff
@@ -119,10 +137,9 @@ If there are any problems with the `sdkdocs-template`, please [raise an issue](h
...
   Conventions for code sample organization, project settings, and naming conventions are described in this document.
   Always test code changes by building and testing the SDK code sample.
-
```
Comment (JohnHake) [inline CONTRIBUTING.md:128]: Superfluous change.

File: CONTRIBUTING.md [hunk 26]
```diff
@@ -10,12 +10,18 @@ For information about contributing to the IntelliJ Platform itself, please visit
...
+Once set up, running the site is as easy as calling `rake preview`.
 
-Alternatively, the site can also be hosted in a [Docker container](https://www.docker.com). On Mac and Windows, this means the site is hosted in a virtual machine. Docker maintains this container, building it based on the instructions in the [`Dockerfile`](Dockerfile). All dependencies (Ruby, etc.) are automatically installed when building the image, which reduces the manual configuration steps. The Docker image is also used to build the [published site](https://www.jetbrains.org/intellij/sdk/docs/index.html), so it is a known working environment.
+Alternatively, the site can also be hosted in a [Docker container](https://www.docker.com).
+On Mac and Windows, this means the site is hosted in a virtual machine.
+Docker maintains this container, building it based on the instructions in the [`Dockerfile`](Dockerfile).
+All dependencies (Ruby, etc.) are automatically installed when building the image, which reduces the manual configuration steps.
+The Docker image is also used to build the [published site](https://www.jetbrains.org/intellij/sdk/docs/index.html), so it is a known working environment.
```
Comment (JohnHake) [inline CONTRIBUTING.md:24]: Correct changes (here and elsewhere in this doc) per our style guide, but these aren't content changes. Is it wiser to postpone style cleanup until the technical content is correct? (IJSDK-252, IJSDK-575)

File: README.md [hunk 27]
```diff
@@ -7,7 +7,7 @@
 
 Welcome to the repository for [IntelliJ Platform SDK Documentation](https://www.jetbrains.org/intellij/sdk/docs/) site.
 
-The code samples can also be found in [IntelliJ SDK Code Samples](https://github.com/JetBrains/intellij-sdk-code-samples) mirror repository. 
+The code samples can also be found in [IntelliJ SDK Code Samples](https://github.com/JetBrains/intellij-sdk-code-samples) mirror repository.
```
Comment (JohnHake) [inline README.md:10]: Superfluous style change

File: CODE_OF_CONDUCT.md [hunk 28]
```diff
@@ -5,5 +5,5 @@ title: Code of Conduct
 
 ## Code of Conduct
 
-This project and the corresponding community is governed by the [JetBrains Open Source and Community Code of Conduct](https://confluence.jetbrains.com/display/ALL/JetBrains+Open+Source+and+Community+Code+of+Conduct). Please make sure you read it. 
-
+This project and the corresponding community is governed by the [JetBrains Open Source and Community Code of Conduct](https://confluence.jetbrains.com/display/ALL/JetBrains+Open+Source+and+Community+Code+of+Conduct).
+Please make sure you read it.
```
Comment (JohnHake) [inline CODE_OF_CONDUCT.md:9]: Superfluous style change. Or is this affecting form or function in some way?

File: _SUMMARY.md [hunk 29]
```diff
@@ -44,7 +44,7 @@
     * [Plugin Configuration File](basics/plugin_structure/plugin_configuration_file.md)
     * [Plugin Logo (Icon)](basics/plugin_structure/plugin_icon_file.md)
     * [Plugin Dependencies](basics/plugin_structure/plugin_dependencies.md)
-* [Dynamic Plugins](basics/plugin_structure/dynamic_plugins.md)    
+* [Dynamic Plugins](basics/plugin_structure/dynamic_plugins.md)
```
Comment (JohnHake) [inline _SUMMARY.md:47]: Superfluous style change.

File: appendix/glossary.md [hunk 30]
```diff
@@ -77,13 +77,13 @@ Stubs
 
 ## V
 Version Control System _(VCS)_
-: The API for [Version Control System](/reference_guide/vcs_integration_for_plugins.md) allows accessing builtin as well as adding custom implementations. 
+: The API for [Version Control System](/reference_guide/vcs_integration_for_plugins.md) allows accessing builtin as well as adding custom implementations.
```
Comment (JohnHake) [inline appendix/glossary.md:80]: Superfluous style change.

File: appendix/glossary.md [hunk 31]
```diff
@@ -77,13 +77,13 @@ Stubs
...
 
 Virtual File _(VF)_
 : A [Virtual File](/basics/architectural_overview/virtual_file.md) represents a file in a &rarr; _Virtual File System_.
 
 Virtual File System _(VFS)_
-: A [Virtual File System](/basics/virtual_file_system.md) provides a unified API for working with files represented as &rarr; _Virtual File_. 
+: A [Virtual File System](/basics/virtual_file_system.md) provides a unified API for working with files represented as &rarr; _Virtual File_.
```
Comment (JohnHake) [inline appendix/glossary.md:86]: Superfluous style change.

File: appendix/plugin_repository_obsolete.md [hunk 32]
```diff
@@ -31,4 +30,3 @@ For your convenience pages previously part of this documentation are linked belo
   * [Plugin Developers List](https://plugins.jetbrains.com/docs/marketplace/plugin-developers-list.html)
 * [Plugin Recommendations](https://plugins.jetbrains.com/docs/marketplace/intellij-plugin-recommendations.html)
 * [Custom Release Channels](https://plugins.jetbrains.com/docs/marketplace/custom-release-channels.html)
-
```
Comment (JohnHake) [inline appendix/plugin_repository_obsolete.md:34]: Superfluous style change.

File: appendix/plugin_repository_obsolete.md [hunk 33]
```diff
@@ -15,10 +15,9 @@ redirect_from:
 ---
 <!-- Copyright 2000-2020 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file. -->
 
-JetBrains provides an official plugins repository [JetBrains Plugin Repository](https://plugins.jetbrains.com) for all IntelliJ Platform-based IDEs, 
-as well as TeamCity (limited functionality).
- 
-> **WARNING** **Part X - Plugin Repository** pages have moved to [JetBrains Marketplace](https://plugins.jetbrains.com/docs/marketplace/about-marketplace.html) documentation, please update your bookmarks. 
+JetBrains provides an official plugins repository [JetBrains Plugin Repository](https://plugins.jetbrains.com) for all IntelliJ Platform-based IDEs, as well as TeamCity (limited functionality).
+
+> **WARNING** **Part X - Plugin Repository** pages have moved to [JetBrains Marketplace](https://plugins.jetbrains.com/docs/marketplace/about-marketplace.html) documentation, please update your bookmarks.
```
Comment (JohnHake) [inline appendix/plugin_repository_obsolete.md:20]: Correct changes per our style guide, but these aren't content changes, so they are superfluous to our users.

**Review by JohnHake — COMMENTED** [review]
Rather than enumerate additional points, I'll leave more general feedback on #advocates-ij-platform.

**Review by JohnHake — CHANGES_REQUESTED** [review]
Thanks for your patience; there are a lot of good improvements here. 
* Evaluating changes to understand if they altered the meaning of a sentence provided some unexpected puzzles.
* I tried to add some brief explanations about my comments...
  * Sometimes synonym substitutions have subtle, unintended changes to meaning.
  * Grammarly is occasionally confused by our writing and offers incorrect suggestions.
  * en-US vs en-GB styles of English
  * And of course, colloquial English.
* I _think_ I have removed all comments about systemic problems (e.g., using "we, you", proper vs. common nouns) into existing or new YT copyediting issues. If I missed a comment specifically about systemic problems, please don't consider it something you should fix for this PR. 
* I tried using the "Insert a suggestion" GitHub review feature so you would have the option to directly accept suggestions as changes in your submission.
* Please, if you don't mind, once you merge PR316 changes into `intellij-sdk-docs/master`, also perform a general maintenance merge from `master` into the `webhelp` branch. (TIA.)

File: tutorials/framework.md [hunk 34]
```diff
@@ -57,15 +55,15 @@ public class DemoFramework extends FrameworkTypeEx {
...
+To make the framework set up available while executing the steps to create a project, the
+`DemoFramework.createProvider()` method must be implemented to return an object of type [`FrameworkSupportInModuleConfigurable`](upsource:///java/idea-ui/src/com/intellij/framework/addSupport/FrameworkSupportInModuleConfigurable.java), which adds the framework to a module.
 In this example the framework is added to any [`ModuleType`](upsource:///platform/lang-api/src/com/intellij/openapi/module/ModuleType.java) without checking, which is usually not the case.
 
 ```java
 @NotNull
 @Override
 public FrameworkSupportInModuleProvider createProvider() {
-    return  new FrameworkSupportInModuleProvider() {
+    retur ew FrameworkSupportInModuleProvider() {
```
[the remark this answers is not in the document]
Comment (YannCebron) [reply tutorials/framework.md:66]: @hsz how did this pass checks on CI?

**Review by YannCebron — COMMENTED** [review]
after handling John's comments and push, please merge back to `webhelp` branch, Thanks!

File: products/clion.md [hunk 35]
```diff
@@ -37,5 +37,5 @@ Test your plugin with versions of CLion you intend to support.
 
 ## Open Source Plugins for CLion
 When learning new APIs, it is helpful to have some representative projects for reference:
-* [C/C+​+​ Coverage](https://github.com/zero9178/C-Cpp-Coverage-for-CLion)
-* [C/C+​+​ Single File Execution](https://github.com/corochann/SingleFileExecutionPlugin)
+* [C/C++ Coverage](https://github.com/zero9178/C-Cpp-Coverage-for-CLion)
+* [C/C++ Single File Execution](https://github.com/corochann/SingleFileExecutionPlugin)
```
[the remark this answers is not in the document]
Comment (hsz) [reply products/clion.md:41]: That was some other plus that looked like a plus sign — the magic of ASCII.

File: basics/ide_development_instance.md [hunk 36]
```diff
@@ -5,24 +5,24 @@ redirect_from:
...
 A version of the JetBrains Runtime is bundled with all IntelliJ Platform-based IDEs.
 To produce accurate results while running or debugging a plugin project in a Development Instance, follow the procedures below to ensure the Development Instance uses a JetBrains Runtime.
 
 ### Determining a JetBrains Runtime Version
 The JetBrains Runtime is determined from the JDK version used to build the plugin project.
-If a plugin is being developed against the Java 8 SE Development Kit 8 for macOS, for example, `jdk-8u212-macosx-x64.dmg`.
+If a plugin is being developed against the Java 8 SE Development Kit 8 for macOS, `jdk-8u212-macosx-x64.dmg`.
 (This example will use macOS, but Windows and Linux follow the same procedure.)
 To acquire the compatible JetBrains Runtime:
 * Go to the [JetBrains Runtime Site](https://confluence.jetbrains.com/display/JBR/JetBrains+Runtime) for general information and the latest build.
```
[the remark this answers is not in the document]
Comment (hsz) [reply basics/ide_development_instance.md:28]: Yan did it by purpose: https://github.com/JetBrains/intellij-sdk-docs/commit/f5e77e1dccc1b37e9eddbc02696c9a3579c8e60f

**Pull request merged**